src/stores/DialogStore/DialogStore.ts

```diff
@@ -20,179 +62,78 @@ export class DialogStore {
         return DialogStore.staticInstance;
     }
 
-    // Region
-    @observable regionDialogVisible: boolean;
-    @action showRegionDialog = () => {
-        this.regionDialogVisible = true;
-    };
-    @action hideRegionDialog = () => {
-        this.regionDialogVisible = false;
-    };
-
-    // Hotkey
-    @observable hotkeyDialogVisible: boolean;
-    @action showHotkeyDialog = () => {
-        this.hotkeyDialogVisible = true;
-    };
-    @action hideHotkeyDialog = () => {
-        this.hotkeyDialogVisible = false;
-    };
-
-    // About
-    @observable aboutDialogVisible: boolean;
-    @action showAboutDialog = () => {
-        this.aboutDialogVisible = true;
-    };
-    @action hideAboutDialog = () => {
-        this.aboutDialogVisible = false;
-    };
-
-    // Preference
-    @observable preferenceDialogVisible: boolean;
-    @action showPreferenceDialog = () => {
-        this.preferenceDialogVisible = true;
-    };
-    @action hidePreferenceDialog = () => {
-        this.preferenceDialogVisible = false;
-    };
-
-    // Layout
-    @observable saveLayoutDialogVisible: boolean;
-    @action showSaveLayoutDialog = (oldLayoutName?: string) => {
-        this.saveLayoutDialogVisible = true;
-        AppStore.Instance.layoutStore.setOldLayoutName(oldLayoutName);
-    };
-    @action hideSaveLayoutDialog = () => {
-        this.saveLayoutDialogVisible = false;
-    };
-
-    // Workspace
-    @observable workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    @action showWorkspaceDialog = (mode = WorkspaceDialogMode.Save) => {
-        this.fileBrowserDialogVisible = false;
-        this.workspaceDialogMode = mode;
-    };
-    @action hideWorkspaceDialog = () => {
-        this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    };
-
-    // Workspace sharing
-    @observable shareWorkspaceDialogVisible: boolean;
-    @action showShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = true;
-    };
-    @action hideShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = false;
+    zIndexManager = AppStore.Instance.zIndexManager;
+
+    @action showDialog = (id: string, props?: showDialogProps) => {
+        if (!this.dialogVisible.get(id)) {
+            switch (id) {
+                case DialogId.Layout:
+                    this.dialogVisible.set(DialogId.Layout, true);
+                    if (props && props.oldLayoutName) {
+                        AppStore.Instance.layoutStore.setOldLayoutName(props.oldLayoutName);
+                    }
+                    this.zIndexManager.assignIndex(DialogId.Layout);
+                    break;
+
+                case DialogId.Workspace:
+                    this.dialogVisible.set(DialogId.FileBrowser, false);
+                    this.workspaceDialogMode = props.mode;
```

Comment:
Add error handlings: `if (props?.mode) {...}`. Same comment for other dialog Ids.

Reply:
Added.

Reply:
We also need error handling for `DialogId.ExternalPage`.

Reply:
okay